editor.planx.uk/src/@planx/components/PropertyInformation/Public.test.tsx

```diff
@@ -0,0 +1,85 @@
+import { MockedProvider } from "@apollo/client/testing";
+import { screen } from "@testing-library/react";
+import React from "react";
+import * as ReactNavi from "react-navi";
+import * as SWR from "swr";
+import { setup } from "testUtils";
+
+import localAuthorityMock from "./mocks/localAuthorityMock";
+import teamMock from "./mocks/teamMock";
+import PropertyInformation from "./Public";
+
+const TEAM = "southwark";
+
+jest.spyOn(SWR, "default").mockImplementation((url: any) => {
+  return {
+    data: url()?.startsWith("https://www.planning.data.gov.uk")
+      ? localAuthorityMock
+      : null,
+  } as any;
+});
+
+jest.spyOn(ReactNavi, "useCurrentRoute").mockImplementation(
+  () =>
+    ({
+      data: {
+        team: TEAM,
+      },
+    } as any)
+);
+
+test("renders a warning for editors if address data is not in state", async () => {
+  const handleSubmit = jest.fn();
+
+  const { user } = setup(
+    <MockedProvider mocks={teamMock} addTypename={false}>
+      <PropertyInformation
+        title="About the property"
+        description="This is the information we currently have about the property"
+        handleSubmit={handleSubmit}
+      />
+    </MockedProvider>
+  );
+
+  expect(screen.getByTestId("error-summary-invalid-graph")).toBeInTheDocument();
+});
+
+test.skip("renders correctly", async () => {
+  const handleSubmit = jest.fn();
+
+  const useStore = jest.fn();
+  useStore.mockReturnValue([
```

Comment:
anyone have guidance/experience on best way to mock `useStore`? i fiddled with a few variations of this that aren't quite working how I expect them to. we do something similar in other components by mocking "previously submitted data", but in this case I want to check state on first render, not after coming "back".